pkg/services/apiserver/storage/entity/storage.go

```diff
@@ -171,11 +171,54 @@ func (s *Storage) Watch(ctx context.Context, key string, opts storage.ListOption
 		Subresource: requestInfo.Subresource,
 	}
 
+	// check for metadata.name field selector
+	if k.Name == "" && opts.Predicate.Field != nil {
+		if v, ok := opts.Predicate.Field.RequiresExactMatch("metadata.name"); ok {
+			// just watch the specific key if we have a name field selector
+			k.Name = v
+		}
+	}
```

Comment:
Should we do the same for `metadata.namespace`? AFAIK we could do something like:
```shell
kubect get --watch=true -A -o json --field-selector='metadata.namespace=somethingElse' thing
```

Reply:
yeah wouldn't hurt, I'll add it